exonum-java-binding/cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/CryptocurrencyServiceImpl.java

```diff
@@ -92,6 +106,74 @@ public void createPublicApiHandlers(Node node, Router router) {
     });
   }
 
+  @Override
+  @TransactionMethod(CREATE_WALLET_TX_ID)
+  public void createWalletTx(TxMessageProtos.CreateWalletTx arguments, TransactionContext context)
+      throws TransactionExecutionException {
+    PublicKey ownerPublicKey = context.getAuthorPk();
+
+    CryptocurrencySchema schema =
+        new CryptocurrencySchema(context.getFork(), context.getServiceName());
+    MapIndex<PublicKey, Wallet> wallets = schema.wallets();
+
+    if (wallets.containsKey(ownerPublicKey)) {
```

Comment:
May also use `checkExecution(!wallets.containsKey(ownerPublicKey), WALLET_ALREADY_EXISTS.errorCode)`, just as the next method.